src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -216,6 +217,9 @@ public async Task<bool> TryGetAsync(ArrayBuilder<DiagnosticData> list, Cancellat
                     using var _2 = ArrayBuilder<AnalyzerWithState>.GetInstance(out var semanticSpanBasedAnalyzers);
                     using var _3 = ArrayBuilder<AnalyzerWithState>.GetInstance(out var semanticDocumentBasedAnalyzers);
 
+                    var telemetryContext = "Pri" + (int)_priorityProvider.Priority;
+                    using var _4 = TelemetryLogging.LogBlockTimeAggregated(FunctionId.RequestDiagnostics_Summary, telemetryContext);
```

Comment:
I am wondering if this will lead to duplication with the telemetry logged in code fix service around the single call to `diagnosticService.GetDiagnosticsForSpanAsync`. The log block there is dedicated for lightbulb Ctrl + Dot path, while the block here is for all the following paths: lightbulb Ctrl + Dot, margin lightbulb background analysis, tagger, all LSP pull diagnostics calls. If we want to log the telemetry here, you definitely need to add more parameterization to the context so we have different blocks for these code paths. The parameters to include would be `_range`, `_incrementalAnalysis`, `_diagnosticKind` (maybe we want to log separate aggregated blocks per-diagnostic kind and also one accumulated block which is agnostic of the diagnostic kind).

Reply:
Oh, I am not sure of the performance penalty we pay by logging multiple aggregated blocks here. If it is expensive, then a single summary block per each code path would be fine, but we definitely need more context than just the priority here.